File: burncare_front/lib/models/auth_response.dart
class AuthResponse {
  final String token;
  final String firstName;
  final String lastName;
  final String email;
  final String role;
  final String profession;

  AuthResponse({
    required this.token,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.role,
    required this.profession,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      token: json['token'] ?? '',
      firstName: json['firstName'] ?? '',
      lastName: json['lastName'] ?? '',
      email: json['email'] ?? '',
      role: json['role'] ?? 'USER',
      profession: json['profession'] ?? '',
    );
  }
}

File: burncare-app/src/main/java/com/burncare/burncare_app/dto/AuthResponse.java
package com.burncare.burncare_app.dto;

import com.burncare.burncare_app.entities.Profession;
import com.burncare.burncare_app.entities.Role;

public record AuthResponse(
        String token,
        String firstName,
        String lastName,
        String email,
        Role role,
        Profession profession
) {}

